publish.go

```diff
@@ -164,3 +168,36 @@ type BatchItemsError []BatchItemResponse
 func (err BatchItemsError) Error() string {
 	return "one or many events may have not been published"
 }
+
+// Format implements fmt.Formatter for BatchItemsError
+func (err BatchItemsError) Format(s fmt.State, verb rune) {
+	switch verb {
+	case 'v':
+		if s.Flag('+') {
+			var messages []string
+			for k, v := range err {
+				messages = append(messages, fmt.Sprintf("[%d]: %+v", k, v))
+			}
+			sort.Strings(messages)
```

Comment:
I think it's at least pointless because the arrays will keep their orders and the messages will start with the correct array keys but to be honest I think it's even broken:
https://play.golang.org/p/5RCvi1YogYd

Reply:
makes sense